parastell/invessel_build.py

```diff
@@ -256,6 +308,144 @@ def generate_components(self):
+                MBTRIs connecting the two ribs.
+        """
+        mb_tris = []
+        for rib_loci_index, _ in enumerate(rib1.rib_loci[0:-1]):
+            corner1 = rib1.mb_verts[rib_loci_index]
+            corner2 = rib1.mb_verts[rib_loci_index + 1]
+            corner3 = rib2.mb_verts[rib_loci_index + 1]
+            corner4 = rib2.mb_verts[rib_loci_index]
+            corners = [corner1, corner2, corner3, corner4]
+            mb_tris += create_moab_tris_from_verts(
+                corners, self.mbc, reverse=reverse
+            )
+        return mb_tris
+
+    def _generate_pymoab_verts(self):
+        """Generate MBVERTEX entities from rib loci in all surfaces"""
+        [
+            surface._generate_pymoab_verts(self.mbc)
+            for surface in self.Surfaces.values()
+        ]
+
+    def _generate_curved_surfaces_pydagmc(self):
+        """Generate the faceted representation of each curved surface and
+        add it to the PyDAGMC model, remembering the surface ids."""
+        self.curved_surface_ids = []
+        for surf_idx, surface in enumerate(self.Surfaces.values()):
+            mb_tris = []
+            for rib, next_rib in zip(surface.Ribs[0:-1], surface.Ribs[1:]):
+                mb_tris += self._connect_ribs_with_tris_moab(
+                    rib, next_rib, reverse=True if surf_idx == 0 else False
+                )
+            dagmc_surface = self.dag_model.create_surface()
+            self.dag_model.mb.add_entities(dagmc_surface.handle, mb_tris)
+            self.curved_surface_ids.append(dagmc_surface.id)
+
+    def _generate_end_cap_surfaces_pydagmc(self):
+        """Generate the faceted representation of the planar end cap surfaces
+        and add them to the PyDAGMC model, remembering the surface ids."""
+        self.end_cap_surface_ids = []
+        for surface, next_surface in zip(
+            list(self.Surfaces.values())[0:-1],
+            list(self.Surfaces.values())[1:],
+        ):
+            for index in (0, -1):
+                mb_tris = self._connect_ribs_with_tris_moab(
+                    surface.Ribs[index],
+                    next_surface.Ribs[index],
+                    reverse=True if index == -1 else False,
+                )
+                end_cap = self.dag_model.create_surface()
+                self.mbc.add_entities(end_cap.handle, mb_tris)
+
+            self.end_cap_surface_ids.append(
+                list(self.dag_model.surfaces_by_id.keys())[-2:]
+            )
+
+    def _generate_volumes_pydagmc(self):
+        """Use the curved surface and end cap surface IDs to build the
+        the volumes by applying the correct surface sense to each surface."""
+        # make the volumes and apply surface sense
+
+        [self.dag_model.create_volume() for _ in list(self.Surfaces)[:-1]]
+
+        # First surface goes to the implicit complement (plasma chamber)
+        self.dag_model.surfaces_by_id[
+            self.curved_surface_ids[0]
+        ].surf_sense = [
```

Comment:
For readability it might be better if this was more like the blocks that follow:
```suggestion
        surface_id = self.curved_surface_ids[0]
        self.dag_model.surfaces_by_id[surface_id].surf_sense = [
```